### ISSUE TITLE
[typescript/no-unused-vars] errors inside declare namespace

### ISSUE THREAD

**Issue by ficristo** [bradzacher/eslint-plugin-typescript#268]
**Repro**
See below

```JSON
{
  "rules": {
    "no-unused-vars": "off",
    "typescript/no-unused-vars": ["error", { "vars": "all", "args": "none" }]
  }
}
```

```TS
export = CodeMirror;
export as namespace CodeMirror;

declare function CodeMirror(host: HTMLElement, options?: CodeMirror.EditorConfiguration): CodeMirror.Editor;

declare namespace CodeMirror {
    function countColumn(line: string, index: number | null, tabSize: number): number;
    var version: string;
}

```

**Expected Result**
No errors.

**Actual Result**
`typescript/no-unused-vars` errors for `function countColumn` and `version`.

**Additional Info**

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `1.0.0-rc.0` |
| `typescript-eslint-parser` | `eslint-plugin-typescript/parser` |
| `typescript`               | `3.1.6` |

**Comment by bradzacher** [comment]
```TS
declare namespace CodeMirror {
    export var Doc : CodeMirror.DocConstructor;
    var version: string; // error: 'version' is defined but never used.
}
```

`var`s are automatically exported in typescript namespaces ([playground](https://www.typescriptlang.org/play/index.html#src=declare%20namespace%20CodeMirror%20%7B%0A%20%20%20%20export%20var%20Doc%20%3A%20string%3B%0A%20%20%20%20var%20version%3A%20string%3B%0A%7D%0A%0ACodeMirror.version.big()%3B%0A)), so version should be marked as defined. This is an easy fix as we can just check the type of the parent.

```TS
interface Object {
    foo(): string,
}
declare var Object: { // error: 'Object' is defined but never used.
    new (value?: any): Object,
    foo(): string
}
```

ambient variable declarations should be marked as used automatically.
Unfortunately the parser doesn't give us a flag for if a variable is marked as `declare` or not ([ast explorer](https://astexplorer.net/#/gist/48223ee9341fd8015f8200e29de2ad1e/cabea38a1d3ed4727dbc44b3cfb01bb8c771c72c)).

Will need to get this added.
This is a low-priority use case though, as not many people use global ambient definitions.
So we won't block 1.0.0 on this.

**Comment by bradzacher** [comment]
from @armano2 https://github.com/bradzacher/eslint-plugin-typescript/issues/266#issuecomment-450930201

> some time ago i added support for declare: JamesHenry/typescript-estree#42
but we are still waiting for eslint/typescript-eslint-parser#584 to be merged before we can use it here
